planner/core/rule_partition_processor.go

```diff
@@ -472,6 +478,33 @@ func partitionRangeForOrExpr(sctx sessionctx.Context, expr1, expr2 expression.Ex
 	return tmp1.union(tmp2)
 }
 
+func partitionRangeForInExpr(sctx sessionctx.Context, args []expression.Expression,
+	pruner *rangePruner) partitionRangeOR {
+	if len(args) <= 1 {
+		return pruner.fullRange()
+	}
+	col, ok := args[0].(*expression.Column)
+	if !ok || col.ID != pruner.col.ID {
+		return pruner.fullRange()
+	}
+
+	var result partitionRangeOR
+	unsigned := mysql.HasUnsignedFlag(col.RetType.Flag)
+	for i := 1; i < len(args); i++ {
+		constExpr, ok := args[i].(*expression.Constant)
+		if !ok {
+			return pruner.fullRange()
+		}
+		val, err := constExpr.Value.ToInt64(sctx.GetSessionVars().StmtCtx)
+		if err != nil {
+			return pruner.fullRange()
+		}
```

Comment:
``` go
if NewIntDatum(val).compareDatum(constExpr) != 0 {
    continue
}
```

Reply:
In create table DDL, we have check that less than XXX must be a constant integer!

Reply:
But this `constExpr` is the argument of `in`...

Reply:
So this one is better?
https://github.com/pingcap/tidb/pull/17210/commits/b780a2c96f7e61f3d83115d234ade8be1f07786d#diff-0411f48b4045359c9f116ef6a779c3efR498

Reply:
If we use EvalInt, it'll be wrong if we call `a in (1, 1.1, 1.1f)`

Reply:
Now I've check the datum.Kind() to be KindUint64 / KindInt64